### ISSUE TITLE
[bun build --compile] Compile-time inlining breaks system paths

### ISSUE THREAD

**Issue by blairdenman** [oven-sh/bun#15049]
### What version of Bun is running?

1.1.34+5e5e7c60f

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

```ts
const APPDATA = process.env.APPDATA;
console.log(APPDATA);
```
Built using `bun build --compile`

### What is the expected behavior?

`C:\Users\<user>\AppData\Roaming` is logged to the console (as per `bun run test.ts`)

### What do you see instead?

`C:Users<user>AppDataRoaming` is logged instead

### Additional information

Temporary workaround:
```ts
const { APPDATA } = process.env;
console.log(APPDATA); // works as expected
```

Similar:
#13063

**Comment by pfgithub** [comment]
This will be fixed by #15062

**Comment by itsBlair-hide** [comment]
Can confirm this is patched and working as expected on the latest canary build. Thanks!
